rules/react.js

```diff
@@ -1,91 +1,146 @@
 module.exports = {
 	"rules": {
 		// Don't allow displayName parameters, but require the component to be named by reference.
-		"react/display-name": [2, {
-			"acceptTranspilerName": true
+		"react/display-name": ["error", {
+			"ignoreTranspilerName": false
 		}],
 		// Allow all prop types at this time.
-		"react/forbid-prop-types": 0,
+		"react/forbid-component-props": "off",
+		// Allow all element types.
+		"react/forbid-elements": "off",
+		// Don't allow messy imports of other component's prop types.
+		"react/forbid-foreign-prop-types": "error",
+		// Warn if the developer is using vague prop type descriptions, like any, array, and object.
+		"react/forbid-prop-types": "warn",
 		// Enforce boolean attributes notation No need for <Component variable={true} />.
 		// Just do <Component variable />.
-		"react/jsx-boolean-value": [2, "never"],
+		"react/jsx-boolean-value": ["error", "never"],
 		// Require closing brackets to be located immediately following the last prop.
-		"react/jsx-closing-bracket-location": [2, "after-props"],
+		"react/jsx-closing-bracket-location": ["error", "after-props"],
 		// Don't allow spaces between curly braces in JSX attributes.
-		"react/jsx-curly-spacing": [2, "never"],
+		"react/jsx-curly-spacing": ["error", "never", {
+			"allowMultiline": true
+		}],
+		// Don't allow spaces before or after the equal sign for components.
+		"react/jsx-equals-spacing": ["error", "never"],
+		// Allow either .js or .jsx files for components.
+		"react/jsx-filename-extension": ["warn", {
+			"extensions": [".js", ".jsx"]
+		}],
+		// The first prop of a component should never be on a new line.
+		"react/jsx-first-prop-new-line": ["error", "never"],
```

Comment:
I think this is in line with our current practices, but I think `multiline` is cleaner. This is a fixable rule, so curious to see what others think.